### PR TITLE
LPD-34912 Fixed table index, and added some enhancements

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/useGetProductByOrderId.ts [hunk 1]
```diff
@@ -9,35 +9,41 @@ import {Liferay} from '../liferay/liferay';
 import HeadlessCommerceDeliveryCatalogImpl from '../services/rest/HeadlessCommerceDeliveryCatalog';
 import HeadlessCommerceDeliveryOrderImpl from '../services/rest/HeadlessCommerceDeliveryOrder';
 
-const useGetProductByOrderId = (orderId: string) => {
-	return useSWR(`/placed-order/${orderId}`, async () => {
-		const placedOrder =
-			await HeadlessCommerceDeliveryOrderImpl.getPlacedOrder(orderId);
+const useGetProductByOrderId = (orderId: string, swrOptions?: any) => {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/hooks/useGetProductByOrderId.ts:12]:
```suggestion
const useGetProductByOrderId = (orderId: string, swrOptions?: SWRConfiguration) => {
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/UninstallModal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+			<p>
+				{i18n.translate(
+					'a-request-to-uninstall-the-license-will-be-processed-and-it-will-no-longer-be-visible-in-your-account'
+				)}
+			</p>
+		</Modal>
+	);
+};
+export default UninstallModal;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/UninstallModal.tsx:76]:
```suggestion

export default UninstallModal;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/UninstallModal.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import ClayButton from '@clayui/button';
+import {useModal} from '@clayui/modal';
+
+import Modal from '../../../../../../../components/Modal';
+import i18n from '../../../../../../../i18n';
+import useProvisioningData from '../hooks/useProvisioningData';
+
+type OrderItem = ReturnType<
+	typeof useProvisioningData
+>['provisioningTableData'][0];
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/UninstallModal.tsx:15]: Move type inside the hook and rename 🥺 

```suggestion
import useProvisioningData, {ProvisioningRow} from '../hooks/useProvisioningData';
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/UninstallModal.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+type UninstallModalProps = {
+	loading: boolean;
+	modal: ReturnType<typeof useModal>;
+	orderItem: OrderItem;
+	uninstall: (orderItem: OrderItem) => void;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/UninstallModal.tsx:21]:
```suggestion
	provisioningRow: ProvisioningRow;
	uninstall: (provisioningRow: ProvisioningRow) => void;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+import i18n from '../../../../../../../i18n';
+import {Liferay} from '../../../../../../../liferay/liferay';
+import consoleOAuth2 from '../../../../../../../services/oauth/Console';
+import ProvisioningDetails from '../components/ProvisioningDetails';
+import {InstallStatus} from '../types';
+import useProvisioningData from './useProvisioningData';
+
+type OrderItem = ReturnType<
+	typeof useProvisioningData
+>['provisioningTableData'][0];
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:22]:
```suggestion
import useProvisioningData, {ProvisioningRow} from './useProvisioningData';
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+						)}
+					</p>
+				),
+				center: true,
+				footer: [
+					<ClayButton
+						className="ml-2 rounded-lg"
+						displayType="unstyled"
+						key="install-cancel-footer"
+						onClick={() => onClose()}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:64]:
```suggestion
						onClick={onClose}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+						size="sm"
+					>
+						{i18n.translate('cancel')}
+					</ClayButton>,
+					undefined,
+					<ClayButton
+						className="ml-2 rounded-lg"
+						displayType="primary"
+						key="install-done-footer"
+						onClick={() => onClose()}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:74]:
```suggestion
						onClick={onClose}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+				],
+				header: i18n.translate('no-cloud-projects-available'),
+				size: 'md' as any,
+			});
+		}
+
+		navigate(`/order/${order?.id}/cloud-provisioning/install`);
+	};
+
+	const uninstall = async (orderItem: OrderItem) => {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:88]:
```suggestion
	const uninstall = async (provisioningRow: ProvisioningRow) => {
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+			installationInProgress:
+				orderItem.status === InstallStatus.IN_PROGRESS,
+			isExpired: orderItem.status === InstallStatus.EXPIRED,
+			isInstalled: orderItem.status === InstallStatus.INSTALLED,
+		};
+	}
+
+	const openUninstallModal = (orderItem: OrderItem) => {
+		setSelectedOrderItem(orderItem);
+		uninstallModal.onOpenChange(true);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:128]:
```suggestion
		setSelectedOrderItem(orderItem);
		
		uninstallModal.onOpenChange(true);
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+	};
+
+	const onOpenDetailsModal = (orderItem: OrderItem) => {
+		const isExpired = orderItem.status === InstallStatus.EXPIRED;
+		const isInstalled = orderItem.status === InstallStatus.INSTALLED;
+		const installationInProgress =
+			orderItem.status === InstallStatus.IN_PROGRESS;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:135]:
```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+					onClose={() => onClose()}
+					orderItem={orderItem}
+				/>
+			),
+			center: true,
+			footer: [
+				undefined,
+				undefined,
+				<div key="details-footer-buttons">
+					{!isInstalled && !isExpired && (
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:155]:
```suggestion
					{readyToInstall && (
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+		}
+
+		setLoading(false);
+	};
+
+	function checkOrderItemStatus(orderItem: OrderItem) {
+		return {
+			inProgress: orderItem.status === InstallStatus.IN_PROGRESS,
+			installationInProgress:
+				orderItem.status === InstallStatus.IN_PROGRESS,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:120]: duplicated

-----

add READY TO INSTALL

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+				<div key="details-footer-buttons">
+					{!isInstalled && !isExpired && (
+						<ClayButton
+							className="border border-primary ml-2 rounded-lg text-primary"
+							disabled={installationInProgress}
+							displayType="secondary"
+							onClick={() => {
+								onClose();
+
+								install();
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:163]:
```suggestion
								onClickInstall();
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 14]
```diff
@@ -0,0 +1,233 @@
...
+							size="sm"
+						>
+							{i18n.translate('uninstall')}
+						</ClayButton>
+					)}
+
+					<ClayButton
+						className="ml-2 rounded-lg"
+						displayType="primary"
+						onClick={() => onClose()}
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:189]:
```suggestion
						onClick={onClose}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningData.ts [hunk 15]
```diff
@@ -74,23 +108,32 @@ const useProvisioningData = (orderId: string) => {
 						productLicenseType
 					),
 					host: '',
-					id: deployment?.id,
+					id: deployment?.id || i,
+					loading: deployment?.loading,
 					orderItem: orderItem.id,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningData.ts:113]:
```suggestion
					orderItemId: orderItem.id,
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
Thanks!

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+
+const useProvisioningActions = ({
+	mutateOrder,
+	order,
+	resourceRequirements,
+	selectedAccount,
+}: UseProvisioningActionsProp) => {
+	const {onClose, onOpenModal} = useModalContext();
+	const [loading, setLoading] = useState<boolean>(false);
+	const [selectedprovisioningRow, setSelectedProvisioningRow] =
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:40]:
```suggestion
	const [selectedProvisioningRow, setSelectedProvisioningRow] =
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+};
+
+const useProvisioningActions = ({
+	mutateOrder,
+	order,
+	resourceRequirements,
+	selectedAccount,
+}: UseProvisioningActionsProp) => {
+	const {onClose, onOpenModal} = useModalContext();
+	const [loading, setLoading] = useState<boolean>(false);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:39]:
```suggestion
	const [loading, setLoading] = useState(false);
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+		navigate(`/order/${order?.id}/cloud-provisioning/install`);
+	};
+
+	const uninstall = async (provisioningRow: ProvisioningRow) => {
+		setLoading(true);
+
+		try {
+			await consoleOAuth2.uninstallApp(order.id, {
+				id: provisioningRow.id,
+				orderItemId: provisioningRow.ProvisioningRowId,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:91]:
```suggestion
				orderItemId: provisioningRow.orderItemId,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+				type: 'danger',
+			});
+		}
+
+		setLoading(false);
+	};
+
+	function checkOrderItemStatus(provisioningRow: ProvisioningRow) {
+		return {
+			installationInProgress:
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:115]:
```suggestion
			inProgress:
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+			isExpired: provisioningRow.status === InstallStatus.EXPIRED,
+			isInstalled: provisioningRow.status === InstallStatus.INSTALLED,
+			readyToInstall:
+				provisioningRow.status === InstallStatus.READY_TO_INSTALL,
+		};
+	}
+
+	const openUninstallModal = (provisioningRow: ProvisioningRow) => {
+		setSelectedProvisioningRow(provisioningRow);
+		uninstallModal.onOpenChange(true);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningActions.tsx:126]:
```suggestion
		setSelectedProvisioningRow(provisioningRow);
		
		uninstallModal.onOpenChange(true);
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningData.ts [hunk 21]
```diff
@@ -74,23 +112,32 @@ const useProvisioningData = (orderId: string) => {
 						productLicenseType
 					),
 					host: '',
-					id: deployment?.id,
-					orderItem: orderItem.id,
+					id: deployment?.id || i,
+					loading: deployment?.loading,
+					ProvisioningRowId: orderItem.id,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/hooks/useProvisioningData.ts:117]:
```suggestion
					orderItemId: orderItem.id,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/ProvisioningDetails.tsx [hunk 22]
```diff
@@ -13,20 +13,16 @@ import useProvisioningData from '../hooks/useProvisioningData';
...
 		image?: string;
 		licenseType?: string;
-		myUserAccount: ReturnType<
-			typeof useMarketplaceContext
-		>['myUserAccount'];
 		name?: string;
 	};
 	onClose: () => void;
-	orderItem?: ReturnType<
+	orderItem: ReturnType<
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/ProvisioningDetails.tsx:25]:
```suggestion
	provisioningRow: ReturnType<
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Pull request closed without merging**